lib/CL/clCreateProgramWithBinary.c

```diff
@@ -85,23 +87,22 @@ POname(clCreateProgramWithBinary)(cl_context                     context,
         "device not found in the device list of the context\n");
     }
   
-  if ((program = (cl_program) malloc (sizeof (struct _cl_program))) == NULL)
+  if ((program = (cl_program) calloc (1, sizeof (struct _cl_program))) == NULL)
```

Comment:
calloc() clears (zero-initializes) the memory, so 1) i don't have to manually set every struct field to 0/NULL, and also 2) allows me to jump to ERROR label and call free() on all pointers in struct, since i can be sure i'll be calling free() on NULL and not some garbage.